add new macros, to avoid using language constructs in translations.

- $PHORUM['file_space_quota'] -> %file_space_quota%
- str_replace(";", ", ", $PHORUM['file_types']) -> %file_types%
- $PHORUM['max_file_size'] -> %max_file_size%

still todo:
- $PHORUM[title] -> %phorum_title%
        "EmailVerifyBody"       =>
        "LostPassEmailSubject"  =>
        "LostPassEmailBody1"    =>
        "LostPassEmailBody2"    =>
        "PMNotifyMessage"       =>
        "PMNotifySubject"       =>
        "RegApprovedEmailBody"  =>
        "VerifyRegEmailBody1"   =>
        "VerifyRegEmailBody2"   =>
- phorum_get_url(PHORUM_LOGIN_URL) -> %login_url%
        "LostPassEmailBody2"    =>
        "RegApprovedEmailBody"  =>
        "VerifyRegEmailBody2"   =>

--- phorum-5.2.15a/include/api/file.php	2010-10-22 00:35:13.428590924 +0300
+++ phorum-5.2.15a/include/api/file.php	2010-10-22 00:46:32.460815378 +0300
@@ -201,9 +201,10 @@
         // Check if the file doesn't exceed the maximum allowed file size.
         if (isset($file["filesize"]) && $PHORUM["max_file_size"] > 0 &&
             $file["filesize"] > $PHORUM["max_file_size"]*1024) {
+            $msg = str_replace("%max_file_size%", $PHORUM['max_file_size'], $PHORUM["DATA"]["LANG"]["FileTooLarge"]);
             return phorum_api_error_set(
                 PHORUM_ERRNO_NOACCESS,
-                $PHORUM["DATA"]["LANG"]["FileTooLarge"]
+                $msg
             );
         }
 
@@ -212,9 +213,10 @@
             $sz = phorum_db_get_user_filesize_total($PHORUM["user"]["user_id"]);
             $sz += $file["filesize"];
             if ($sz > $PHORUM["file_space_quota"]*1024) {
+                $msg = str_replace("%file_space_quota%", $PHORUM['file_space_quota'], $PHORUM["DATA"]["LANG"]["FileOverQuota"]);
                 return phorum_api_error_set(
                     PHORUM_ERRNO_NOACCESS,
-                    $PHORUM["DATA"]["LANG"]["FileOverQuota"]
+                    $msg,
                 );
             }
         }
@@ -236,9 +238,10 @@
 
             // Check if the extension for the file is an allowed extension.
             if (!in_array($ext, $allowed_exts)) {
+                $msg = str_replace('%file_types%', str_replace(";", ", ", $PHORUM['file_types']), $PHORUM["DATA"]["LANG"]["FileWrongType"]);
                 return phorum_api_error_set(
                     PHORUM_ERRNO_NOACCESS,
-                    $PHORUM["DATA"]["LANG"]["FileWrongType"]
+                    $msg
                 );
             }
         }
--- phorum-5.2.15a/include/controlcenter/files.php	2010-10-22 00:35:13.265257620 +0300
+++ phorum-5.2.15a/include/controlcenter/files.php	2010-10-22 00:42:47.237295406 +0300
@@ -101,7 +101,11 @@
 }
 
 if($PHORUM["file_types"]){
-    $PHORUM["DATA"]["FILE_TYPE_LIMIT"]=$PHORUM["DATA"]["LANG"]["FileTypeLimits"];
+    $PHORUM["DATA"]["FILE_TYPE_LIMIT"]= str_replace(
+      '%file_types%', 
+      str_replace(";", ", ", $PHORUM['file_types']),
+      $PHORUM["DATA"]["LANG"]["FileTypeLimits"]
+    );
 }
 
 if($PHORUM["file_space_quota"]){
